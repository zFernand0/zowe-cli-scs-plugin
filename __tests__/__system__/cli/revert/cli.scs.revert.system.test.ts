/*
 * This program and the accompanying materials are made available under the terms of the
 * Eclipse Public License v2.0 which accompanies this distribution, and is available at
 * https://www.eclipse.org/legal/epl-v20.html
 *
 * SPDX-License-Identifier: EPL-2.0
 *
 * Copyright Contributors to the Zowe Project.
 */

import { ITestEnvironment } from "../../../__src__/environment/doc/response/ITestEnvironment";
import { TestEnvironment } from "../../../__src__/environment/TestEnvironment";
import { runCliScript } from "../../../__src__/TestUtils";

let TEST_ENV: ITestEnvironment;

describe("secure-credential-store revert command", () => {
    beforeAll(async () => {
    // Create test environment without installing the plugin
        TEST_ENV = await TestEnvironment.setUp({testName: "revert_command"});

        // Install the plugin
        await TestEnvironment.installPlugin(TEST_ENV);
    });

    afterAll(async () => {
        await TestEnvironment.cleanUp(TEST_ENV);
    });

    it("should display the help", async () => {
        const response = await runCliScript(__dirname + "/__scripts__/revert_help.sh", TEST_ENV);
        expect(response.stderr.toString()).toBe("");
        expect(response.stdout.toString()).toMatchSnapshot();
        expect(response.status).toBe(0);
    });

    it("should fail without for-sure option", async () => {
        const response = await runCliScript(__dirname + "/__scripts__/invalid_command.sh", TEST_ENV);
        expect(response.stdout.toString()).toBe("");
        expect(response.stderr.toString()).toMatchSnapshot();
        expect(response.status).toBe(1);
    });

    it("should fail if secure credential manager not enabled", async () => {
        const response = await runCliScript(__dirname + "/__scripts__/invalid_credmgr.sh", TEST_ENV);
        expect(response.stdout.toString()).toBe("");
        expect(response.stderr.toString()).toMatchSnapshot();
        expect(response.status).toBe(0);
    });
});
