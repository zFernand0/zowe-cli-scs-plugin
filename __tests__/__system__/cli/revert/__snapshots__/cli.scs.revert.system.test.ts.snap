// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`secure-credential-store revert command should display the help 1`] = `
"
 COMMAND NAME
 ------------

   revert | r

 DESCRIPTION
 -----------

   Reverts all secure profiles to be stored in plain text

 USAGE
 -----

   zowe secure-credential-store revert [options]

 REQUIRED OPTIONS
 ----------------

   --for-sure  | -f (boolean)

      Specify this option to confirm that you want to revert all credentials to be
      stored insecurely.

 GLOBAL OPTIONS
 --------------

   --response-format-json  | --rfj (boolean)

      Produce JSON formatted data from a command

   --help  | -h (boolean)

      Display help text

   --help-examples  (boolean)

      Display examples for all the commands in a group

   --help-web  | --hw (boolean)

      Display HTML help in browser

{
  \\"success\\": true,
  \\"exitCode\\": 0,
  \\"message\\": \\"The help was constructed for command: revert.\\",
  \\"stdout\\": \\"\\\\n COMMAND NAME\\\\n ------------\\\\n\\\\n   revert | r\\\\n\\\\n DESCRIPTION\\\\n -----------\\\\n\\\\n   Reverts all secure profiles to be stored in plain text\\\\n\\\\n USAGE\\\\n -----\\\\n\\\\n   zowe secure-credential-store revert [options]\\\\n\\\\n REQUIRED OPTIONS\\\\n ----------------\\\\n\\\\n   --for-sure  | -f (boolean)\\\\n\\\\n      Specify this option to confirm that you want to revert all credentials to be\\\\n      stored insecurely.\\\\n\\\\n GLOBAL OPTIONS\\\\n --------------\\\\n\\\\n   --response-format-json  | --rfj (boolean)\\\\n\\\\n      Produce JSON formatted data from a command\\\\n\\\\n   --help  | -h (boolean)\\\\n\\\\n      Display help text\\\\n\\\\n   --help-examples  (boolean)\\\\n\\\\n      Display examples for all the commands in a group\\\\n\\\\n   --help-web  | --hw (boolean)\\\\n\\\\n      Display HTML help in browser\\\\n\\\\n\\",
  \\"stderr\\": \\"\\",
  \\"data\\": \\"\\\\n COMMAND NAME\\\\n ------------\\\\n\\\\n   revert | r\\\\n\\\\n DESCRIPTION\\\\n -----------\\\\n\\\\n   Reverts all secure profiles to be stored in plain text\\\\n\\\\n USAGE\\\\n -----\\\\n\\\\n   zowe secure-credential-store revert [options]\\\\n\\\\n REQUIRED OPTIONS\\\\n ----------------\\\\n\\\\n   --for-sure  | -f (boolean)\\\\n\\\\n      Specify this option to confirm that you want to revert all credentials to be\\\\n      stored insecurely.\\\\n\\\\n GLOBAL OPTIONS\\\\n --------------\\\\n\\\\n   --response-format-json  | --rfj (boolean)\\\\n\\\\n      Produce JSON formatted data from a command\\\\n\\\\n   --help  | -h (boolean)\\\\n\\\\n      Display help text\\\\n\\\\n   --help-examples  (boolean)\\\\n\\\\n      Display examples for all the commands in a group\\\\n\\\\n   --help-web  | --hw (boolean)\\\\n\\\\n      Display HTML help in browser\\\\n\\\\n\\"
}"
`;

exports[`secure-credential-store revert command should fail if secure credential manager not enabled 1`] = `
"Error:
Invalid credential manager setting

Possible Causes:
  Expected secure credential manager to be enabled

Resolutions:
  Run \\"zowe config set CredentialManager @zowe/secure-credential-store-for-zowe-cli\\"

"
`;

exports[`secure-credential-store revert command should fail without for-sure option 1`] = `
"
Syntax Error:
Missing Required Option:
--for-sure (-f)

Option Description:
Specify this option to confirm that you want to revert all credentials to be
stored insecurely.

Use \\"zowe secure-credential-store revert --help\\" to view command description, usage, and options.
"
`;
